### PR TITLE
Denis/intrawallet tx asset

### PR DESCRIPTION
**Pull request by Nebyt**

### CODE REVIEW

File: tests/screenObjects/walletHistory.screen.ts [hunk 1]
```diff
@@ -1,6 +1,48 @@
...
+export const assetsListComponent = () => driver.$('//*[@resource-id="assetList"]')
+export const assetItemSelector = '//*[@resource-id="assetItem"]'
+export const tokenNameSelector = '//*[@resource-id="tokenInfoText"]'
+export const tokenFingerPrintSelector = '//*[@resource-id="tokenFingerprintText"]'
+export const tokenAmountSelector = '//*[@resource-id="tokenAmount"]'
 
 export const isDisplayed = async () => {
-  return (await sendButton().isDisplayed) && (await receiveButton().isDisplayed)
+  return (
+    (await sendButton().isDisplayed) &&
```
Comment (stackchain) [inline tests/screenObjects/walletHistory.screen.ts]:
```suggestion
    (await sendButton().isDisplayed()) &&
```

File: tests/screenObjects/walletHistory.screen.ts [hunk 2]
```diff
@@ -1,6 +1,48 @@
...
+    (await receiveButton().isDisplayed) &&
```
Comment (stackchain) [inline tests/screenObjects/walletHistory.screen.ts]:
```suggestion
    (await receiveButton().isDisplayed()) &&
```

File: tests/screenObjects/walletHistory.screen.ts [hunk 3]
```diff
@@ -1,6 +1,48 @@
...
+    (await transactionsTabButton().isDisplayed()) &&
+    (await assetsTabButton().isDisplayed())
+  )
+}
+
+export const getAllTransactions = async () => {
+  return await driver.$$(txHistoryListItemSelector)
+}
```
Comment (stackchain) [inline tests/screenObjects/walletHistory.screen.ts]:
```suggestion
export const getAllTransactions = async () => driver.$$(txHistoryListItemSelector)
```

File: tests/screenObjects/walletHistory.screen.ts [hunk 4]
```diff
@@ -1,6 +1,48 @@
...
+
+export const getDateHeaders = async () => {
+  return await driver.$$(txsDayHeaderTextSelector())
+}
```
Comment (stackchain) [inline tests/screenObjects/walletHistory.screen.ts]:
```suggestion
export const getDateHeaders = async () => driver.$$(txsDayHeaderTextSelector())
```

File: tests/screenFunctions/walletHistory.screenFunctions.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+  // copy address
+  const receiverAddress = await copyFirstUnusedAddress()
+  // go back to the transactions screen
+  await driver.back()
+  return receiverAddress
+}
+
+export const copyFirstUnusedAddress = async (): Promise<string> => {
+  return await unusedAddressRows().$(addressTextSelector).getText()
+}
```
Comment (stackchain) [inline tests/screenFunctions/walletHistory.screenFunctions.ts]:
```suggestion
export const copyFirstUnusedAddress = async (): Promise<string> => unusedAddressRows().$(addressTextSelector).getText()
```

**Comment by stackchain** [comment]
`formatWithToken` signature has changed; some merge conflicts will raise.

File: tests/helpers/utils.ts [hunk 6]
```diff
@@ -35,90 +6,38 @@ export async function isElementChecked(element: WebdriverIO.Element): Promise<bo
...
-  try {
-    isErrorDisplayed = await errorModal
-        .errorView()
-        .waitForDisplayed({timeout: DEFAULT_TIMEOUT, interval: DEFAULT_INTERVAL})
-  } catch (e) {
-    // nothing to do, an error didn't appear
-    isErrorDisplayed = false
+export const amPmTo24 = (inputStringTime) => {
+  const ampmPart = inputStringTime.split(' ')[1]
+  const timePart = inputStringTime.split(' ')[0]
```
Comment (zuzunker) [inline tests/helpers/utils.ts]:
```suggestion
  const [timePart, ampmPart] = inputStringTime.split(' ')
```

File: tests/helpers/utils.ts [hunk 7]
```diff
@@ -35,90 +6,38 @@ export async function isElementChecked(element: WebdriverIO.Element): Promise<bo
...
+  const hours = parseInt(timePart.split(':')[0])
+  const minutes = timePart.split(':')[1]
+  const seconds = timePart.split(':')[2]
```
Comment (zuzunker) [inline tests/helpers/utils.ts]:
```suggestion
  const [hoursStr, minutes, seconds] = timePart.split(':')
  const hours = parseInt(hoursStr, 10)
```

File: tests/screenObjects/walletHistory.screen.ts [hunk 8]
```diff
@@ -1,6 +1,44 @@
...
 export const receiveButton = () => driver.$('//*[@resource-id="receiveButton"]')
+export const transactionsTabButton = () => driver.$('//*[@resource-id="transactionsTabButton"]')
+export const assetsTabButton = () => driver.$('//*[@resource-id="assetsTabButton"]')
+export const balanceText = () => driver.$('//*[@resource-id="balanceText"]')
+export const pairedTotalText = () => driver.$('//*[@resource-id="pairedTotalText"]')
+
+// transactions list
+export const txsHistoryListComponent = () => driver.$('//*[@resource-id="txHistoryList"]')
+export const txsEmptyHistoryComponent = () => driver.$('//*[@resource-id="emptyHistoryComponent"]')
+export const txsDayHeaderTextSelector = () => '//*[@resource-id="dayHeaderText"]'
```
Comment (stackchain) [inline tests/screenObjects/walletHistory.screen.ts]:
```suggestion
export const txsDayHeaderTextSelector = '//*[@resource-id="dayHeaderText"]'
```

File: tests/screenObjects/walletHistory.screen.ts [hunk 9]
```diff
@@ -1,6 +1,44 @@
...
+    (await sendButton().isDisplayed()) &&
+    (await receiveButton().isDisplayed()) &&
+    (await transactionsTabButton().isDisplayed()) &&
+    (await assetsTabButton().isDisplayed())
+  )
+}
+
+export const getAllTransactions = async () => driver.$$(txHistoryListItemSelector)
+
+export const getDateHeaders = async () => driver.$$(txsDayHeaderTextSelector())
```
Comment (stackchain) [inline tests/screenObjects/walletHistory.screen.ts]:
```suggestion
export const getDateHeaders = async () => driver.$$(txsDayHeaderTextSelector)
```

File: tests/screenObjects/createWalletScreens/createWalletCredentials.screen.ts [hunk 10]
```diff
@@ -3,12 +3,9 @@ import {WALLET_ALREADY_EXISTS, MUST_BE_FILLED} from '../../helpers/errors'
...
-// @ts-ignore
-export const walletNameEdit = (): WebdriverIO.Element => getAllEditFields()[0]
+export const walletNameEdit = () => getAllEditFields()[0]
 export const walletNameExistsError = () => driver.$(`[text="${WALLET_ALREADY_EXISTS}"]`);
 export const walletNameIsEmptyError = () => driver.$(`text="${MUST_BE_FILLED}"`);
-// @ts-ignore
-export const spendingPasswordEdit = (): WebdriverIO.Element => getAllEditFields()[1]
-// @ts-ignore
-export const repeatSpendingPasswordEdit = (): WebdriverIO.Element => getAllEditFields()[2]
+export const spendingPasswordEdit = () => getAllEditFields()[1]
```
Comment (stackchain) [inline tests/screenObjects/createWalletScreens/createWalletCredentials.screen.ts:9]: hey @Nebyt can we select the fields directly, avoiding index accessing?

File: tests/screenObjects/createWalletScreens/createWalletCredentials.screen.ts [hunk 11]
```diff
@@ -3,12 +3,9 @@ import {WALLET_ALREADY_EXISTS, MUST_BE_FILLED} from '../../helpers/errors'
...
+export const repeatSpendingPasswordEdit = () => getAllEditFields()[2]
```
Comment (stackchain) [inline tests/screenObjects/createWalletScreens/createWalletCredentials.screen.ts:10]: ⬆️

File: tests/screenObjects/createWalletScreens/createWalletCredentials.screen.ts [hunk 12]
```diff
@@ -3,12 +3,9 @@ import {WALLET_ALREADY_EXISTS, MUST_BE_FILLED} from '../../helpers/errors'
 const getAllEditFields = () => driver.$$('//android.widget.EditText')
 
 export const credentialsView = () => driver.$('//*[@resource-id="credentialsView"]')
-// @ts-ignore
-export const walletNameEdit = (): WebdriverIO.Element => getAllEditFields()[0]
+export const walletNameEdit = () => getAllEditFields()[0]
```
Comment (stackchain) [inline tests/screenObjects/createWalletScreens/createWalletCredentials.screen.ts:6]: ⬇️

**Review by stackchain — CHANGES_REQUESTED** [review]

File: tests/screenObjects/createWalletScreens/createWalletCredentials.screen.ts (hunk 10 above)
Comment (Nebyt) [reply tests/screenObjects/createWalletScreens/createWalletCredentials.screen.ts:9]: Unfortunately, I didn't find another way.

File: tests/screenObjects/createWalletScreens/createWalletCredentials.screen.ts (hunk 11 above)
Comment (Nebyt) [reply tests/screenObjects/createWalletScreens/createWalletCredentials.screen.ts:10]: Unfortunately, I didn't find another way.

File: tests/screenObjects/createWalletScreens/createWalletCredentials.screen.ts (hunk 12 above)
Comment (Nebyt) [reply tests/screenObjects/createWalletScreens/createWalletCredentials.screen.ts:6]: Unfortunately, I didn't find another way.

File: tests/screenFunctions/walletHistory.screenFunctions.ts [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+  expect(
+    (await getAssetByName(tokenName, walletHistoryScreen.assetItemSelector, walletHistoryScreen.tokenNameSelector))
+      .success,
+    `The token "${tokenName} is not found in assets"`,
+  ).to.be.true
+}
+
+export const getAllAssets = async (assetItemSelector: string) => {
+  return await driver.$$(assetItemSelector)
+}
```
Comment (stackchain) [inline tests/screenFunctions/walletHistory.screenFunctions.ts]:
```suggestion
export const getAllAssets = async (assetItemSelector: string) => driver.$$(assetItemSelector)
```

**Review by wolverineks — APPROVED** [review]

**Pull request merged**